### PR TITLE
Bump kubernetes-client-bom from 6.10.0 to 6.11.0

### PR DESCRIPTION
**Pull request by manusa**
Kubernetes Client 6.11.0 was just released: https://github.com/fabric8io/kubernetes-client/releases/tag/v6.11.0

/cc @metacosm

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

**Comment by gastaldi** [comment]
Waiting on @metacosm's feedback before merging this PR

**Comment by metacosm** [comment]
The update seems OK for us based on https://github.com/quarkiverse/quarkus-operator-sdk/actions/runs/8436245950/job/23103338631

**Pull request merged**
